tests/test_plugins/test_exceptions.py

```diff
@@ -0,0 +1,7 @@
+from starlette_context.errors import StarletteContextError
+
+
+def test_context_exception_is_catchable():
```

Comment:
I don't think this test is sufficient. Maybe consider adding something like

```python

# test_plugin.py

@pytest.mark.asyncio
async def test_exception_plugin_can_be_handled(
    mocked_request: Request, plugin: plugins.Plugin
):
    with mock.patch.object(plugin, 'extract_value_from_header_by_key', side_effect=StarletteContextError):
        try:
            await plugin.extract_value_from_header_by_key(mocked_request)
        except Exception as e:
            # todo: make some assertions here?
            pass
```
This way we can test a real scenario. You could move this `assert issubclass` there.

Reply:
I've added a test in the requested style in `test_plugin.py`. I do think it is valuable to have an actual try/catch or pytest.raises() so that its clear why this change is necessary.

I do however think the `mock.patch` followed by immediately calling that patched function, is effectively just raising the exception directly with more lines of code.

Reply:
I kind of agree. Only one of them is necessary. One is simple, the other one slightly complicated but also serves as a documentation and having both doesn't hurt. As I said, it's just a minor nitpick.

Thank you!